plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/nessie/IcebergNessieCatalogModule.java

```diff
@@ -43,14 +45,19 @@ protected void setup(Binder binder)
 
     @Provides
     @Singleton
-    public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalogConfig icebergNessieCatalogConfig)
+    public static NessieIcebergClient createNessieIcebergClient(IcebergNessieCatalogConfig config)
     {
-        return new NessieIcebergClient(
-                HttpClientBuilder.builder()
-                        .withUri(icebergNessieCatalogConfig.getServerUri())
-                        .build(NessieApiV1.class),
-                icebergNessieCatalogConfig.getDefaultReferenceName(),
-                null,
-                ImmutableMap.of());
+        HttpClientBuilder builder = HttpClientBuilder.builder()
+                .withUri(config.getServerUri())
+                .withDisableCompression(!config.isCompressionEnabled());
+
+        config.getReadTimeoutMillis().ifPresent(builder::withReadTimeout);
+        config.getConnectTimeoutMillis().ifPresent(builder::withConnectionTimeout);
+        config.getAuthenticationType().ifPresent(type -> {
+            if (type.equals(BEARER)) {
+                config.getBearerToken().ifPresent(token -> builder.withAuthentication(BearerAuthenticationProvider.create(token)));
```

Comment:
Could we utilize `builder.fromConfig(...)`?

Reply:
we don't have a properties map here from the user input and I think maybe building a map here just for calling `builder.fromConfig(...)` is an overkill.

Reply:
thx - LGTM :+1: